tests/test_tools.py

```diff
@@ -2009,3 +2010,155 @@ def test_remove_duplicate_docs(inp, expected):
     actual = filter_employment_for_advisees(inp[0], inp[1], "ms", dt.date(2021,6,3))
     assert actual == expected
+
+person1 = {"_id":"tstark", "aka":"iron man", "name":"tony stark"}
+person2 = {"_id":"nromanov", "aka":"black widow", "name":"natasha romanov"}
+PEOPLE = [person1, person2]
+
+presentation1 = {"_id":"abc", "authors":["tstark"], "begin_year":2018, "begin_month":1, "begin_day":1,
+                 "department":"apam", "institution":"columbiau", "status":"accepted", "type":"award"}
+presentation2 = {"_id":"ghi", "authors":["tstark","nromanov"], "begin_year":2019, "begin_month":1, "begin_day":2,
+                 "end_year":2019, "end_month":1, "end_day":8, "department":"physics", "institution":"rutgersu",
+                 "status":"cancelled", "type":"poster"}
+PRESENTATIONS = [presentation1, presentation2]
+
+institution1 = {"_id":"columbiau", "city":"New York", "country":"USA", "name":"Columbia University", "state":"NY"}
+institution2 = {"_id":"rutgersu", "city":"New Brunswick", "country":"USA", "name":"Rutgers University", "state":"NJ"}
+INSTITUTIONS = [institution1, institution2]
+
+@pytest.mark.parametrize(
+    "args, kwargs, expected",[
+
+    #this tests no kwargs
+    ([PEOPLE, PRESENTATIONS, INSTITUTIONS, "tstark"],
+    {},
+    [{"_id": "abc",
+    "authors": "tony stark",
+    "begin_day_suffix": "st",
+    "begin_year": 2018,
+    "begin_month": 1,
+    "begin_day": 1,
+    "date": dt.date(2018, 1, 1),
+    "day_suffix": "st",
+    "department": {'name': 'apam'},
+    "institution": {'city': 'New York',
+                   'country': 'USA',
+                   'name': 'Columbia University',
+                   'state': 'NY'},
+    "status": "accepted",
+    "type": "award"}]
+    ),
+
+    #this tests 'statuses' kwarg
+    ([PEOPLE, PRESENTATIONS, INSTITUTIONS, "tstark"],
+    {"statuses" : ["all"]},
+    [{"_id": "ghi",
+    "authors": "tony stark, natasha romanov",
+    "begin_day_suffix": "nd",
+    "begin_year": 2019,
+    "begin_month": 1,
+    "begin_day": 2,
+    "date": dt.date(2019, 1, 2),
+    "day_suffix": "nd",
+    "end_year": 2019,
+    "end_month": 1,
+    "end_day": 8,
+    "end_day_suffix": "th",
+    "department": {'name': 'physics'},
+    "institution": {'city': 'New Brunswick',
+                   'country': 'USA',
+                   'name': 'Rutgers University',
+                   'state': 'NJ'},
+    "status": "cancelled",
+    "type": "poster"},
+    {"_id": "abc",
+    "authors": "tony stark",
+    "begin_day_suffix": "st",
+    "begin_year": 2018,
+    "begin_month": 1,
+    "begin_day": 1,
+    "date": dt.date(2018, 1, 1),
+    "day_suffix": "st",
+    "department": {'name': 'apam'},
+    "institution": {'city': 'New York',
+                   'country': 'USA',
+                   'name': 'Columbia University',
+                   'state': 'NY'},
+    "status": "accepted",
+    "type": "award"}]
+    ),
+
+    #this tests 'statuses' and 'types' kwargs together
```

Comment:
it seems as if you are thinking of tests that test your testing structure (args, kwargs etc.) but we want to focus on testing the function.  Roughly speaking, look for all the `if` statements in the function and make sure you have a test that enters each of those.  Try and do it in as few tests as possible.

Just fyi, this is backwards from how we want to do things normally, so I am teaching you bad habits, but given the task I have given you, this is the strategy.  The RIGHT way to do it is to think what you want your function to do, write tests that test for that behavior, THEN write the function.....